### PR TITLE
Photocopier now uses ink cartridges'

### LINKED ISSUE #8083: Photocopier needs to use toner cartridges.
### Description

#7944 added toner cartridges to the game for use in airlock painters. The photocopier is also supposed to use toner cartridges when printing documents. Photocopier.cs will need to be reworked to hold and consume charges from toner cartridges.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #8083

CL: [New] Photocopiers now consume ink from toner cartridges to print paper.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs [hunk 1]
```diff
@@ -26,13 +24,35 @@ public class Photocopier : NetworkBehaviour, ICheckedInteractable<HandApply>, IS
...
+		private IEnumerator Start()
+		{
+			//We can't spawn stuff into slots on Awake() because it will screech at us
+			//We do this on Start() and after a frame when the game loads.
+			yield return WaitFor.EndOfFrame;
+			if (tonerPrefab == null) yield break;
+			foreach (var slot in inkStorage.GetItemSlots())
+			{
+				if(slot.IsOccupied) continue;
+				Inventory.ServerSpawnPrefab(tonerPrefab, slot);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: Can't you do this with inventory populaters or specify directly in the prefab what to spawn in the item storage?

File: UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs [hunk 2]
```diff
@@ -26,13 +24,35 @@ public class Photocopier : NetworkBehaviour, ICheckedInteractable<HandApply>, IS
...
+			}
+			var cartradgeSlot = inkStorage.GetTopOccupiedIndexedSlot();
+			inkCartadge = cartradgeSlot.ItemObject.GetComponent<Toner>();
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: then to skip this you could just make it so inkCartadge => cartradgeSlot.ItemObject.GetComponent<Toner>();

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: with some ? in case of nulls

**Comment by MaxIsJoe** [comment]
Done. Review pls

File: UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs [hunk 3]
```diff
@@ -122,6 +133,11 @@ public void ServerPerformInteraction(HandApply interaction)
 			}
 		}
 
+		private void RemoveInkCartrdge()
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]:
```suggestion
		private void RemoveInkCartridge()
```

File: UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs [hunk 4]
```diff
@@ -7,16 +7,14 @@
...
 		public NetTabType NetTabType;
 		public int trayCapacity;
 
-		[SyncVar]
-		private Internal.Printer printer;
+		[SyncVar] private Internal.Printer printer;
 
-		[SyncVar]
-		private Internal.Scanner scanner;
+		[SyncVar] private Internal.Scanner scanner;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: can we syncvar these objects? I think Mirror only supported basic types and Unity objects

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: It seems to be already working so im not going to change it

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: I didn't say it isn't working. I said it won't be a problem until it becomes one and then we will have to read profiles and rediscover this to solve the issue. Always cache the result of get component when you can.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: Ah shit, I read the wrong thing

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: I'm glad im not the only one who misreads stuff here lol

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: Whats' your say on this @corp-0 ? is it good to go?

**Review by corp-0 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs (hunk 4 above)
Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: We don't want to do this. These objects will be instantiated on every sync, according to Mirror: https://mirror-networking.gitbook.io/docs/guides/data-types (`Classes` bullet point).

**Comment by Bod9001** [comment]
looks good just waiting on requested changes

**Comment by MaxIsJoe** [comment]
merge pls

**Comment by Bod9001** [comment]
"if (printer.CanAddPageToTray(interaction.HandObject)) return true;
				if (scanner.CanPlaceDocument(interaction.HandObject)) return true;"
				
I assume you don't need to synchronise data for this?

**Comment by MaxIsJoe** [comment]
You dont

File: UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs (hunk 4 above)
Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: Didn't realise the code was already syncing those classes. Would've let it slide but good to get it fixed, cheers. So it still works on client/server?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Bureaucracy/Photocopier.cs]: I have said this before, this is not my code; it works as it is and I didn't do anything except change the format of where the attribute is.

**Review by Bod9001 — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]

**Pull request merged**